packages/ember-glimmer/lib/renderer.js

```diff
@@ -40,6 +40,49 @@ class DynamicScope {
   }
 }
 
+class RootState {
+  constructor(root, env, template, self, parentElement, dynamicScope) {
+    assert(`You cannot render \`${self.value()}\` without a template.`, template);
+
+    this.id = getViewId(root);
+    this.root = root;
+    this.result = undefined;
+    this.shouldReflush = false;
+
+    let options = this.options = {
+      alwaysRevalidate: false
+    };
+
+    this.render = () => {
+      let result = this.result = template.asEntryPoint().render(self, env, {
+        appendTo: parentElement,
+        dynamicScope
+      });
+
+      // override .render function after initial render
+      this.render = () => {
+        result.rerender(options);
+      };
+    };
```

Comment:
Indeed, but then we have to keep ahold of all of the other values needed for initial render. Also, it would require a much larger function to be optimized when 99.99999% of the time we only want the smaller version (since we only initial render once).